module/apmelasticsearch/client_test.go

```diff
@@ -303,6 +306,116 @@ func TestDestination(t *testing.T) {
 	test("http://[2001:db8::1]:80/_search", "2001:db8::1", 80)
 }
 
+func TestTraceHeaders(t *testing.T) {
+	headers := make(map[string]string)
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		for k, vs := range req.Header {
+			headers[k] = strings.Join(vs, " ")
+		}
+	}))
+	defer server.Close()
+	client := &http.Client{Transport: apmelasticsearch.WrapRoundTripper(http.DefaultTransport)}
+
+	req, err := http.NewRequest("GET", server.URL, nil)
+	require.NoError(t, err)
+
+	_, _, _ = apmtest.WithTransaction(func(ctx context.Context) {
+		_, err := client.Do(req.WithContext(ctx))
+		assert.NoError(t, err)
+	})
+
+	assert.Contains(t, headers, apmhttp.ElasticTraceparentHeader)
+	assert.Contains(t, headers, apmhttp.W3CTraceparentHeader)
+	assert.Contains(t, headers, apmhttp.TracestateHeader)
+}
+
+func TestClientSpanDropped(t *testing.T) {
```

Comment:
@axw this and `TestClientTransactionUnsampled` are copied from `apmhttp/client_test.go`. They do verify that we're setting the headers when a span is dropped and when a transaction is not sampled, but what do you think about paring them back to just verifying we have the headers? I'm happy either way since this is essentially doubling up on checking the calculation of the headers, vs. just making sure we're setting them.

Reply:
I think what you have here is good. I do think it's important to check the trace context of the transaction vs. span is used like you have in this test.